packages/inventory/src/types.ts

```diff
@@ -9,3 +10,35 @@ export type StocktakeRow = Pick<
   | 'stocktakeNumber'
   | 'status'
 >;
+
+export interface StocktakeLine extends StocktakeLineNode {
+  isCreated?: boolean;
+  isDeleted?: boolean;
+  isUpdated?: boolean;
+}
+
+export interface StocktakeItem {
+  id: string;
+  itemCode: string;
+  itemName: string;
+  isDeleted?: boolean;
+  lines: StocktakeLine[];
+}
+
+export interface Stocktake extends Omit<StocktakeNode, 'lines' | '__typename'> {
+  lines: StocktakeLine[];
+}
+
+export interface StocktakeController extends Omit<Stocktake, 'lines'> {
+  lines: StocktakeItem[];
+  update: (key: string, value: string) => void;
+}
```

Comment:
Hmm.

Wasn't quite sure here. Normally we have `RequisitionNode` -> `Requisition` -> `CustomerRequisition` and the same for invoices. But here it's `StocktakeNode` -> `Stocktake` -> `StocktakeController` which isn't great naming

Reply:
I've carefully considered and have managed to convince myself that the naming is fine 😁 
Basically, I have no better suggestions